dstep/translator/MacroIndex.d

```diff
@@ -20,66 +21,63 @@ import dstep.translator.Preprocessor;
 
 class MacroIndex
 {
-    private static bool pred(in Cursor a, in Cursor b)
-    {
-        return a.location.lexicalLess(b.location);
-    }
-
     private TranslationUnit translUnit;
-    private alias CursorRedBlackTree =
-        RedBlackTree!(Cursor, (a, b) => pred(a, b));
-    private CursorRedBlackTree expansions;
-    private Cursor[string] definitions;
-
+    private bool delegate (Cursor, Cursor) lessOp;
+    private Cursor[] expansions;
+    private Cursor[string] globalCursors_;
     private Directive[] directives;
 
-    private static string uniqueID(in Cursor cursor)
-    {
-        import std.format : format;
-        return format(
-            "%s@%s:%d",
-            cursor.spelling,
-            cursor.location.path,
-            cursor.location.offset);
-    }
-
     this(TranslationUnit translUnit)
     {
         this.translUnit = translUnit;
 
-        expansions = new CursorRedBlackTree();
-
-        Cursor[string] recent;
+        auto expansionsAppender = appender!(Cursor[])();
 
         foreach (cursor, parent; translUnit.cursor.all)
         {
             if (cursor.kind == CXCursorKind.CXCursor_MacroExpansion)
-            {
-                expansions.insert(cursor);
-
-                Cursor* def = cursor.spelling in recent;
+                expansionsAppender.put(cursor);
 
-                if (def !is null)
-                    definitions[uniqueID(cursor)] = *def;
-            }
-            else if (cursor.kind == CXCursorKind.CXCursor_MacroDefinition)
-            {
-                recent[cursor.spelling] = cursor;
-            }
+            if (!cursor.spelling.empty)
+                globalCursors_[extendedSpelling(cursor)] = cursor;
         }
 
+        lessOp = translUnit.relativeCursorLocationLessOp();
+        expansions = expansionsAppender.data.sort!((a, b) => lessOp(a, b)).array;
```

Comment:
No, it can't. It is private member of MacroIndex and is inaccessible from inside of `sort`.